### PR TITLE
Add github pipeline | update gems and appraisals

### PR DESCRIPTION
**Pull request by ttstarck**
### Changes
1. Added github pipeline to run unit tests against mutliple rails versions and gem versions.
2. Updated Appraisals file to have the four combinations of sidekiq 5 & 6 and rails 5 and 6.
3. Updated gemspec to increase minimum version cap max version of dependency gems.
4. Other general cleanups.

### CODE REVIEW

File: .github/workflows/pipeline.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        ports:
+          - 6379:6379
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [2.7, 3.0, 3.1]
+        gemfile:
+          - Gemfile
+          - gemfiles/rails_5_sidekiq_5.gemfile
+#          - gemfiles/rails_5_sidekiq_6.gemfile
```
Comment (ttstarck) [inline .github/workflows/pipeline.yml:25]: These tests fail, I've created https://invoca.atlassian.net/browse/TECH-10254 to address these issues.

File: .github/workflows/pipeline.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          - gemfiles/rails_6_sidekiq_5.gemfile
+#          - gemfiles/rails_6_sidekiq_6.gemfile
```
Comment (ttstarck) [inline .github/workflows/pipeline.yml:27]: We'll re-enable in https://invoca.atlassian.net/browse/TECH-10254

File: .github/workflows/pipeline.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    services:
+      redis:
+        image: redis
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+        ports:
+          - 6379:6379
```
Comment (ttstarck) [inline .github/workflows/pipeline.yml:17]: This lets us run our tests against redis instead of mocking redis.

File: sidekiq-superworker.gemspec [hunk 4]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
   s.license       = 'MIT'
   s.metadata["allowed_push_host"] = "https://gem.fury.io/invoca"
 
-  s.add_dependency 'sidekiq', '>= 2.1.0'
-  s.add_dependency 'activesupport', '>= 3.2'
-  s.add_dependency 'activemodel', '>= 3.2'
+  s.add_dependency 'sidekiq', '>= 5.0', '< 6'
+  s.add_dependency 'activesupport', '>= 5.0', '< 7'
+  s.add_dependency 'activemodel', '>= 5.0', '< 7'
```
Comment (ttstarck) [inline sidekiq-superworker.gemspec:18]: There isn't any need to support any versions less than 5 for any of these.

File: sidekiq-superworker.gemspec [hunk 5]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |s|
   s.license       = 'MIT'
   s.metadata["allowed_push_host"] = "https://gem.fury.io/invoca"
 
-  s.add_dependency 'sidekiq', '>= 2.1.0'
-  s.add_dependency 'activesupport', '>= 3.2'
-  s.add_dependency 'activemodel', '>= 3.2'
+  s.add_dependency 'sidekiq', '>= 5.0', '< 6'
```
Comment (ttstarck) [inline sidekiq-superworker.gemspec:16]: We'll bump this to `< 7` in https://invoca.atlassian.net/browse/TECH-10254

File: spec/subjob_processor_spec.rb [hunk 6]
```diff
@@ -0,0 +1,81 @@
+require 'spec_helper'
+
+describe Sidekiq::Superworker::SubjobProcessor do
```
Comment (ttstarck) [inline spec/subjob_processor_spec.rb:3]: This was a missing test that I created a long time ago that I forgot to check in.

File: spec/support/superworkers.rb [hunk 7]
```diff
@@ -6,7 +6,7 @@
 
     sidekiq_options :queue => 'sidekiq_superworker_test'
 
-    def perform
+    def perform(*args)
```
Comment (ttstarck) [inline spec/support/superworkers.rb:9]: Was causing small test failure in higher sidekiq version.

File: Appraisals [hunk 8]
```diff
@@ -1,10 +1,22 @@
 appraise "default" do
 end
 
-appraise "rails_3" do
-  gem "rails", "3.2.19", require: true
+appraise "rails_5_sidekiq_5" do
```
Comment (ttstarck) [inline Appraisals:4]: We can potentially stop supporting rails 5 soon but for now I think we'll keep it around.

**Review by phurley67 — APPROVED** [review]
👍🏻

**Pull request merged**